pipeline/test_load.py

```diff
@@ -1 +1,328 @@
+    get_or_insert_tags(mock_cursor, "new_tag", album_id=1)
+    calls = [
+        call("SELECT tag_id FROM tag WHERE name = %s", ("new_tag",)),
+        call("INSERT INTO tag(name) VALUES (%s) RETURNING tag_id", ("new_tag",)),
+        call(
+            "INSERT INTO album_tag_assignment(tag_id, album_id) VALUES (%s, %s)", (
+                1, 1)
+        ),
+    ]
+    mock_cursor.execute.assert_has_calls(calls)
+
+
+@patch("load.get_cursor")
+def test_get_or_insert_tags_existing(mock_get_cursor):
+    """Test case for get_or_insert_tags with existing tag."""
+    mock_cursor = MagicMock()
+    mock_get_cursor.return_value = mock_cursor
+    mock_cursor.fetchone.return_value = [1]
+    get_or_insert_tags(mock_cursor, "existing_tag", album_id=1)
+    calls = [
+        call("SELECT tag_id FROM tag WHERE name = %s", ("existing_tag",)),
+        call(
+            "INSERT INTO album_tag_assignment(tag_id, album_id) VALUES (%s, %s)", (
+                1, 1)
+        ),
+    ]
+    mock_cursor.execute.assert_has_calls(calls)
+
+
+@patch("load.get_cursor")
+def test_insert_album_or_track_purchase_album(mock_get_cursor):
+    """Test case for insert_album_or_track_purchase with album purchase."""
+    mock_cursor = MagicMock()
+    mock_get_cursor.return_value = mock_cursor
+    insert_album_or_track_purchase(
+        mock_cursor, "timestamp", 10.0, 1, album_id=1)
+    mock_cursor.execute.assert_called_with(
+        "INSERT INTO album_purchase(album_id, timestamp, amount_usd, country_id)\
+                VALUES (%s, %s, %s, %s)",
+        (1, "timestamp", 10.0, 1),
+    )
+
+
+@patch("load.get_cursor")
+def test_insert_album_or_track_purchase_track(mock_get_cursor):
+    """Test case for insert_album_or_track_purchase with track purchase."""
+    mock_cursor = MagicMock()
+    mock_get_cursor.return_value = mock_cursor
+    insert_album_or_track_purchase(
+        mock_cursor, "timestamp", 10.0, 1, track_id=1)
+    mock_cursor.execute.assert_called_with(
+        "INSERT INTO track_purchase(track_id, timestamp, amount_usd, country_id)\
+                VALUES (%s, %s, %s, %s)",
+        (1, "timestamp", 10.0, 1),
+    )
+
+
+@patch("load.get_cursor")
+def test_insert_album_sale(mock_get_cursor):
+    """Test case for insert_album_sale function."""
+    mock_cursor = MagicMock()
+    mock_get_cursor.return_value = mock_cursor
+    album_sale = {
+        "utc_date": 1718881844.178138,
+        "artist_name": "FINAL",
+        "item_type": "a",
+        "item_description": "Infinite Guitar 2",
+        "currency": "USD",
+        "amount_paid": 2.0,
+        "item_price": 0.0,
+        "amount_paid_usd": 2.0,
+        "country": "United Kingdom",
+        "art_id": 3978634151,
+        "url": "https://final1.bandcamp.com/album/infinite-guitar-2",
+        "artist_url": "https://final1.bandcamp.com",
+        "country_code": "gb",
+        "amount_paid_fmt": "$2",
+        "art_url": "https://f4.bcbits.com/img/a3978634151_7.jpg",
```

Comment:
It mostly looks good, but have you considered using classes for tests? It's easier to dupllicate data if needed

Reply:
I did, but I'm not too familiar with using classes with pytest, and considering the time constraints, it would be best to implement that another time.

Reply:
Fair enough